### DISCUSSION TITLE
将 Supabase 作为下一个后端服务 - 愧怍的小站

### DISCUSSION THREAD

**Discussion by giscus** [kuizuo/blog#196] [General]
# 将 Supabase 作为下一个后端服务 - 愧怍的小站

本文介绍了如何使用 Supabase 作为后端服务，使开发人员可以更快地构建和部署应用程序，无需配置数据库或编写复杂的身份验证代码。将使用 Nuxt.js 和 Supabase，以实现一个图床网站为例，来演示如何在前端中使用 Supabase API 和 Storage 服务。

https://kuizuo.cn/use-supabase-as-backend-service#comments

**Comment by mylysddp** [comment]
这里有一个疑问，前端如果没有设计用户账户登录机制（不需要登录有些接口就允许使用），在没有身份鉴别的情况下，如果用户看到了前端代码里面的supbase url，理论上是不是可以直接改表里面数据。这时候行级锁也控制不住吧。

**Reply by kuizuo** [reply]
supabase 是有两个key，一个是 public (公开)，另一个 secret (私密) ,
![image](https://user-images.githubusercontent.com/61005888/224612721-d2b35117-45c2-498e-a1ef-b3a562dbcb9d.png)

其实对应的 key 下的描述已经很明朗了，这里我给你翻译一下

public
> 如果已为表和配置的策略启用了行级安全性，则在浏览器中使用此密钥是安全的。


secret 
> 此密钥具有绕过行级安全性的能力。永远不要公开共享它。
 
所以即便用户拿到了 前端 代码里面的supbase url，也就是 `https://azlbliyjwcemojkwazto.supabase.co` 与 `public key`（通常是协议头 apikey），只要表数据有设置行级锁是肯定能够控制的。如果没设置的话自然就是谁都能修改公开表的数据。

![image](https://user-images.githubusercontent.com/61005888/224613589-dc87acb7-49ce-4959-8134-53f36103007d.png)


当然还有一个情况就是 用户拿到了 `secret key`，这就不难理解，拥有该 key 相当于拥有超级管理员权限。

**Reply by mylysddp** [reply]
了解了，多谢！

**Comment by sunzehui** [comment]
Prisma有试过吗，可以配合supabase

**Reply by kuizuo** [reply]
你想说的应该就是使用prisma来连接supabase的数据库，到supabase的控制台中找到远程连接的postgrasql，配置下prisma就可以操作数据库了

**Comment by NeutrinoLiu** [comment]
挺细的介绍，mongodb难民表示要从头再来了。
ref: https://www.reddit.com/r/mongodb/comments/1fct01v/fuck_you_mongodb/
